garak/resources/autodan/autodan.py

```diff
@@ -205,7 +205,8 @@ def autodan_generate(
             best_new_adv_prefix = new_adv_prefixes[best_new_adv_prefix_id]
 
             adv_prefix = best_new_adv_prefix
-            success, gen_str = check_for_attack_success(adv_prefix, test_prefixes)
+            gen_str = generator.generate(prompt=adv_prefix)[0]
```

Comment:
Is it worth avoiding multiple response generation here? by e.g. pushing `generator.generations`/setting `generator.generations=1`/calling `generate()`/popping `generator.generations`?

This is a situation and solution pattern I see in a few places - maybe `generate()` is the place to allow an override for `generations`, or even other things (rather than the `generator` class constructor)

Reply:
That seems right to me, yes. I just worry that if we set `generator.generaions=1` and the probe is being run as part of a series (for some reason -- this is a lot of time on a single probe anyway), it may have downstream effects. Definitely worth allowing an override for `generation` in the `generate()`method.

Reply:
will open an issue to track that - #585